### PR TITLE
ci: use shared workflows

### PR DESCRIPTION
**Pull request by TheUnderScorer**

### CODE REVIEW

File: .github/workflows/docs-and-coverage.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+name: Generate docs and coverage report
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  generate-docs-and-coverage:
```
[the remark this answers is not in the document]
Comment (ilfa) [reply .github/workflows/docs-and-coverage.yml:9]:
```yml
   with:
      prepare-gh-pages-commands: |
        mv docs/* ./gh-pages
        mv coverage/lcov-report ./gh-pages/coverage
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -1,32 +1,12 @@
...
+    branches-ignore:
+      - main
+    paths-ignore:
+      - '**.md'
+
 jobs:
-  build:
-    name: Lint, build and test
-    runs-on: ubuntu-latest
-    timeout-minutes: 3
```
Comment (ilfa) [inline .github/workflows/test.yml:9]: I'm curious why we had this, can it be the reason for the stucked workflow?
image

Comment (TheUnderScorer) [reply .github/workflows/test.yml:9]: Yep! We'll need to change the branch protection settings to match name of the new action for tests.

**Review by ilfa — APPROVED** [review]

**Pull request merged**